File: packages/typescript/src/core/providers/static.ts
import type { BaseWallet } from '../base.js'
import { WalletProvider } from './base.js'

export class StaticWalletProvider extends WalletProvider {
  constructor(private readonly wallet: BaseWallet) {
    super()
  }

  async getActiveWallet(): Promise<BaseWallet> {
    return this.wallet
  }
}

File: packages/python/src/agent_wallet/core/providers/static.py
"""Static single-wallet provider implementation."""

from __future__ import annotations

from agent_wallet.core.base import BaseWallet
from agent_wallet.core.providers.base import WalletProvider


class StaticWalletProvider(WalletProvider):
    """Provider for a single in-memory wallet resolved from environment."""

    def __init__(self, wallet: BaseWallet) -> None:
        self._wallet = wallet

    async def get_active_wallet(self) -> BaseWallet:
        return self._wallet
